src/west/commands/project.py

```diff
@@ -24,6 +24,83 @@
 # branch.
 _MANIFEST_REV_BRANCH = 'manifest-rev'
 
+class Reset(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'reset',
+            _wrap('''
+            Reset the repositories to a specific revision for the project given.
+            Also reset other remote repositories to the related revision found in
+            the manifest file.
+            Additional git arguments, such as --hard can be provided using
+            --git-args.'''))
+
+    def do_add_parser(self, parser_adder):
+        return _add_parser(
+            parser_adder, self,
+            _arg('-r', '--revision',
+                 dest='revision',
+                 metavar='REVISION',
+                 required=True,
+                 help='''ToBeWritten'''),
+            _arg('-g', '--git-args',
+                 dest='gitargs',
+                 metavar='GIT_ARGS',
+                 default='',
+                 required=False,
+                 help='''Additional git arguments to be passed with 'git reset,'
+                         e.g '--hard' '''),
+            _arg('projects', metavar='PROJECT', nargs='?', default='zephyr')
+            )
+
+    def do_run(self, args, user_args):
+        # Arg parse, if project not given, use 'zephyr'
+        # Only one project can be given.
+        # ToDo: Check SHA is valid ?
+        n=0
+        args.projects = [args.projects]
+        project = _projects(args)[0]
+        manifest = _special_project(args, 'manifest')
+        reset_projects = _all_projects(args)
+
+        while True:
+            n += 1
+
+            sha = project.sha or project.revision
+            rev_list = _git(project,
+                   'rev-list --topo-order {} ^{}'.format(sha, args.revision),
```

Comment:
Also I think this is clearer: `git rev-list --topo-order sha1..sha2`

Reply:
And do you care about `--topo-order`? you don't use the commits themselves

Reply:
> Can you document what this does exactly? "list the commits reachable from the project's current SHA but not by the revision that we are trying to reset to"?

correct, and when that list ends-up empty, we know which manifest version to use, and thus which other repo revision to reset to.
Will add comment.

>Also I think this is clearer: git rev-list --topo-order sha1..sha2

`git rev-list --topo-order sha1..sha2` is short-hand for `sha1 ^sha2`.
I don't mind changing to short-hand version, but we might need to switch back to current version, in case we need to support resetting to SHAs on other branches.

> And do you care about --topo-order

from code point of view: no
but it was nice to use on command line and during testing of code, as the list of revisions are printed in same order, as you see with `git log`
Can remove if the `--topo-order` makes is harder to understand code.